src/fklearn/validation/validator.py

```diff
@@ -21,6 +21,7 @@ def validator_iteration(data: pd.DataFrame,
                         train_fn: LearnerFnType,
                         eval_fn: EvalFnType,
                         predict_oof: bool = False,
+                        return_eval_fn_on_train: bool = False,
```

Comment:
just a slight nitpick, I'd name it `return_eval_logs_on_train`,  since what you're returing is the logs of the evaluation and not the evaluation function